### DISCUSSION TITLE
"Bad Signature" error for images hosted on camo.githubusercontent.com and linked from Markdown

### DISCUSSION THREAD

**Discussion by simonw** [community/community#102194] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I'm seeing a frustrating bug with Markdown that I've generated using the [/markdown API endpoint](https://docs.github.com/en/rest/markdown/markdown).

Image syntax like `![A pisco sour in a fancy crystal glass, with a thin layer of white foam on top](https://static.simonwillison.net/static/2022/pisco-sour.jpg)` that is processed by that API comes up as HTML that looks like this:

`<p><a target="_blank" rel="noopener noreferrer nofollow" href="https://camo.githubusercontent.com/ffd38778207720a016f107646a85073036e52b3837874f3dd86cd34a6a57d3f0/68747470733a2f2f7374617469632e73696d6f6e77696c6c69736f6e2e6e65742f7374617469632f323032322f706973636f2d736f75722e6a7067"><img src="https://camo.githubusercontent.com/ffd38778207720a016f107646a85073036e52b3837874f3dd86cd34a6a57d3f0/68747470733a2f2f7374617469632e73696d6f6e77696c6c69736f6e2e6e65742f7374617469632f323032322f706973636f2d736f75722e6a7067" alt="A pisco sour in a fancy crystal glass, with a thin layer of white foam on top" data-canonical-src="https://static.simonwillison.net/static/2022/pisco-sour.jpg" style="max-width: 100%;"></a></p>`

This would be OK... except I've started seeing the following error message from some of these images that I processed a while ago:

https://camo.githubusercontent.com/6ac29b2022a4cd0d49a12a97177f2eed894b9b2c1facf0ed7e7e18d2f638201f/68747470733a2f2f7374617469632e73696d6f6e77696c6c69736f6e2e6e65742f7374617469632f323032322f706973636f2d736f75722e6a7067 - returns "Bad Signature"

I'd like to understand what's going on here.

Is this a measure to prevent people from long term hot-linking to images on `camo.githubusercontent.com`? If so that's fine, but it should be documented somewhere so I know to avoid this pattern.

Or is it some other bug?

Two of my own issues exploring this further:
- https://github.com/simonw/datasette.io/issues/156
- https://github.com/simonw/til/issues/88

**Comment by naveen-98** [comment]
It seems like you're encountering a "Bad Signature" error for images hosted on camo.githubusercontent.com when linked from Markdown generated using the /markdown API endpoint. The issue may be related to the link's signature, which helps prevent long-term hot-linking.

Here's a solution:

1. **Update Image Links:**
   - Instead of directly using the camo.githubusercontent.com link, try updating your Markdown to use the original image link without going through the camo proxy. For example:

     Change from:
     ```markdown
     ![Alt text](https://camo.githubusercontent.com/...)
     ```

     To:
     ```markdown
     ![Alt text](https://static.simonwillison.net/static/2022/pisco-sour.jpg)
     ```

2. **Check GitHub Documentation:**
   - Review GitHub's documentation on camo.githubusercontent.com or image proxy behavior to see if there are any specific restrictions or recommendations regarding long-term usage. It's possible that GitHub has implemented measures to prevent certain types of usage, and this information may not be explicitly stated in the API documentation.

3. **Reach Out to GitHub Support:**
   - If the issue persists and you cannot find a solution in the documentation, consider reaching out to GitHub Support for clarification. They may provide insights into the "Bad Signature" error and offer guidance on how to handle image links in your specific use case.

**Comment by geechartier** [comment]
Did you ever find a solution, please? I am having the same problem. I don't even need the images to be copied to GitHub and sourced from there, but I cannot find an option to turn that off.
